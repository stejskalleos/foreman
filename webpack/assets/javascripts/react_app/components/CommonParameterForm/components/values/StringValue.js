import React from 'react';
import PropTypes from 'prop-types';

const StringValue = ({
  id,
  name,
  css,
  value,
  onChange,
  fullScreen,
  isMasked,
}) => (
  <textarea
    id={id}
    name={name}
    rows={fullScreen ? 30 : 1}
    className={`${css} ${isMasked && 'masked-input'}`}
    value={value}
    onChange={e => onChange(e.target.value)}
  />
);

StringValue.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  css: PropTypes.string,
  value: PropTypes.any,
  onChange: PropTypes.func.isRequired,
  fullScreen: PropTypes.bool,
  isMasked: PropTypes.bool,
};

StringValue.defaultProps = {
  id: '',
  css: '',
  fullScreen: false,
  value: '',
  isMasked: false,
};

export default StringValue;
