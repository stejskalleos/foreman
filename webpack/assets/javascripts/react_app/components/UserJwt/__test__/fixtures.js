import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
