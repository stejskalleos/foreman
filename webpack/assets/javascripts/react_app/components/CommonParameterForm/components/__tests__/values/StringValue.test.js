import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { stringValueFixtures } from '../fixtures';
import StringValue from '../../values/StringValue';


describe('StringValue', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(StringValue, stringValueFixtures));
});
