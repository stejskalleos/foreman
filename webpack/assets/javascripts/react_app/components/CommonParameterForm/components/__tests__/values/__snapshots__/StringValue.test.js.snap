// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringValue rendering with fullScreen 1`] = `
<textarea
  className="form-control false"
  id="common_parameter_value"
  name="common_parameter[value]"
  onChange={[Function]}
  rows={30}
  value="something"
/>
`;

exports[`StringValue rendering without fullScreen 1`] = `
<textarea
  className="form-control false"
  id="common_parameter_value"
  name="common_parameter[value]"
  onChange={[Function]}
  rows={1}
  value="something"
/>
`;
